module/channel/src/Infrastructure/Handler/DeleteChannelCommandHandler.php

```diff
@@ -10,27 +10,56 @@
 namespace Ergonode\Channel\Infrastructure\Handler;
 
 use Ergonode\Channel\Domain\Command\DeleteChannelCommand;
+use Ergonode\Channel\Domain\Entity\AbstractChannel;
+use Ergonode\Channel\Domain\Query\ExportQueryInterface;
 use Ergonode\Channel\Domain\Repository\ChannelRepositoryInterface;
+use Ergonode\EventSourcing\Infrastructure\Bus\CommandBusInterface;
+use Ergonode\Exporter\Infrastructure\Provider\RemoveExportArtifactsCommandFactoryProvider;
+use Ergonode\ExporterFile\Domain\Entity\FileExportChannel;
 use Webmozart\Assert\Assert;
 
 class DeleteChannelCommandHandler
 {
-    private ChannelRepositoryInterface $repository;
+    private ChannelRepositoryInterface $channelRepository;
 
-    public function __construct(ChannelRepositoryInterface $repository)
-    {
-        $this->repository = $repository;
+    private ExportQueryInterface $exportQuery;
+
+    private RemoveExportArtifactsCommandFactoryProvider $removeExportArtifactsCommandFactoryProvider;
+
+    private CommandBusInterface $commandBus;
+
+    public function __construct(
+        ChannelRepositoryInterface $channelRepository,
+        ExportQueryInterface $exportQuery,
+        RemoveExportArtifactsCommandFactoryProvider $removeExportArtifactsCommandFactoryProvider,
+        CommandBusInterface $commandBus
+    ) {
+        $this->channelRepository = $channelRepository;
+        $this->exportQuery = $exportQuery;
+        $this->removeExportArtifactsCommandFactoryProvider = $removeExportArtifactsCommandFactoryProvider;
+        $this->commandBus = $commandBus;
     }
 
     /**
      * @throws \Exception
      */
-    public function __invoke(DeleteChannelCommand $command): void
+    public function __invoke(DeleteChannelCommand $deleteChannelCommand): void
     {
-        $channel = $this->repository->load($command->getId());
+        $channel = $this->channelRepository->load($deleteChannelCommand->getId());
+
+        Assert::isInstanceOf(
+            $channel,
+            AbstractChannel::class,
+            sprintf('Can\'t find channel "%s"', $deleteChannelCommand->getId()->getValue())
+        );
 
-        Assert::notNull($channel, sprintf('Can\'t fid channel "%s"', $command->getId()->getValue()));
+        $exportIds = $this->exportQuery->getExportIdsByChannelId($channel->getId());
 
-        $this->repository->delete($channel);
+        $this->channelRepository->delete($channel);
+        $commandFactory = $this->removeExportArtifactsCommandFactoryProvider->provide(FileExportChannel::TYPE);
```

Comment:
No, no, no. In channel module use `Ergonode\ExporterFile\Domain\Entity\FileExportChannel`. 

A better idea is sending the event `RemovedExpotEvent` to EventBus, and Handlers in others export module. 

This idea only works with File export :(